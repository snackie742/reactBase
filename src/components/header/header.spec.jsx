import React from 'react';
import { shallow } from 'enzyme';
import Header from './header';

const defaultProps = {
};

const setup = (propOverrides) => {
  const props = Object.assign({}, defaultProps, propOverrides);
  const wrapper = shallow(<Header {...props} />);
  const component = wrapper.instance();
  return { wrapper, component };
};

describe('Header', () => {
  describe('<Header/>', () => {
    describe('(Snapshot)', () => {
      test('renders as expected', () => {
        const { wrapper } = setup(defaultProps);
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});

