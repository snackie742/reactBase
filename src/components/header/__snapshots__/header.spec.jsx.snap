// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header <Header/> (Snapshot) renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row"
>
        <div
                className="col-sm-3 col-lg-3 col-xl-2"
        >
                <img
                        alt="logo"
                        className="App-logo"
                        src="mlb.png"
                />
        </div>
        <div
                className="col-sm-9"
        >
                <h1
                        className="App-title"
                >
                        MLB Scoreboard
                </h1>
        </div>
</div>,
      "className": "App-header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-sm-3 col-lg-3 col-xl-2"
>
            <img
                        alt="logo"
                        className="App-logo"
                        src="mlb.png"
            />
</div>,
          <div
            className="col-sm-9"
>
            <h1
                        className="App-title"
            >
                        MLB Scoreboard
            </h1>
</div>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="logo"
              className="App-logo"
              src="mlb.png"
/>,
            "className": "col-sm-3 col-lg-3 col-xl-2",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "App-logo",
              "src": "mlb.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="App-title"
>
              MLB Scoreboard
</h1>,
            "className": "col-sm-9",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MLB Scoreboard",
              "className": "App-title",
            },
            "ref": null,
            "rendered": "MLB Scoreboard",
            "type": "h1",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
>
          <div
                    className="col-sm-3 col-lg-3 col-xl-2"
          >
                    <img
                              alt="logo"
                              className="App-logo"
                              src="mlb.png"
                    />
          </div>
          <div
                    className="col-sm-9"
          >
                    <h1
                              className="App-title"
                    >
                              MLB Scoreboard
                    </h1>
          </div>
</div>,
        "className": "App-header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-sm-3 col-lg-3 col-xl-2"
>
              <img
                            alt="logo"
                            className="App-logo"
                            src="mlb.png"
              />
</div>,
            <div
              className="col-sm-9"
>
              <h1
                            className="App-title"
              >
                            MLB Scoreboard
              </h1>
</div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="logo"
                className="App-logo"
                src="mlb.png"
/>,
              "className": "col-sm-3 col-lg-3 col-xl-2",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "App-logo",
                "src": "mlb.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1
                className="App-title"
>
                MLB Scoreboard
</h1>,
              "className": "col-sm-9",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "MLB Scoreboard",
                "className": "App-title",
              },
              "ref": null,
              "rendered": "MLB Scoreboard",
              "type": "h1",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
